filesystem/ios/Plugin/Filesystem.swift

```diff
@@ -0,0 +1,156 @@
+import Foundation
+
+@objc public class Filesystem: NSObject {
+
+    public enum FilesystemError: String, Error {
+        case noParentFolder = "Parent folder doesn't exist"
+        case noSave = "Unable to save file"
+        case failEncode = "Unable to encode data to utf-8"
+        case noAppend = "Unable to append file"
+        case notEmpty = "Folder is not empty"
+    }
```

Comment:
This will not work as expected. While it can conform to the `Error` protocol, calling `localizedDescription` on it will cause it to be bridged to an `NSError` and the `RawRepresentable` string will get lost. Instead of the defined string value it will print a generated, inscrutable description of the object. This should conform to LocalizedError to make sure the human-readable values are passed back.
```suggestion
    public enum FilesystemError: LocalizedError {
        case noParentFolder, noSave, failEncode, noAppend, notEmpty
        
        public var errorDescription: String? {
            switch self {
            case .noParentFolder:
                return "Parent folder doesn't exist"
            case .noSave:
                return "Unable to save file"
            case .failEncode:
                return "Unable to encode data to utf-8"
            case .noAppend:
                return "Unable to append file"
            case .notEmpty:
                return "Folder is not empty"
            }
        }
    }
```

Reply:
yeah, on the Plugin class when it gets caught, I send the raw value to JS, not the localized description

Reply:
Ah, I missed the separate handling. But that could be simplified. Instead of multiple `catch` statements, it could use one for any type of error and call `localizedDescription` on it.